src/utils_lua.c

```diff
@@ -274,38 +235,170 @@ int luaC_pushvalue(lua_State *L, value_t v, int ds_type) /* {{{ */
   return 0;
 } /* }}} int luaC_pushvalue */
 
-int luaC_pushvaluelist(lua_State *L, const data_set_t *ds,
-                       const value_list_t *vl) /* {{{ */
+static int luaC_pushlabelset(lua_State *L, label_set_t label) /* {{{ */
 {
+  /*
+    label_set_t will be mapped to the following Lua table:
+
+    label = {
+      [1] => { name => value },
+      ...
+      [N] => { ... }
+    }
+   */
   lua_newtable(L);
+  label_pair_t *ptr = label.ptr;
+  for (int i = 0; i < label.num; i++) {
+    lua_newtable(L);
+    lua_pushstring(L, ptr->value);
+    lua_setfield(L, -2, ptr->name);
+    /* label[N] = {} */
+    lua_rawseti(L, -2, i + 1);
+    ptr++;
+  }
+  return 0;
+} /* }}} int luaC_pushlabelset */
 
-  lua_pushstring(L, vl->host);
-  lua_setfield(L, -2, "host");
+static int luaC_pushmetadata(lua_State *L, meta_data_t *meta) /* {{{ */
+{
+  /*
+    meta_data_t will be mapped to the following Lua table:
+
+    meta => {
+      [1] => {
+        key => ...,
+        value => ...,
+      },
+      ...
+      [N] => {
+        ...
+      }
+    }
```

Comment:
As with labels: metadata keys are unique, and I think we should simplify this to:

```lua
meta => {
  key0 => "value0",
  key1 => 42,
  …
}
```

Reply:
Fixed it.